jujugui/static/gui/src/app/app.js

```diff
@@ -1187,6 +1196,11 @@ YUI.add('juju-gui', function(Y) {
           showPay={window.juju_config.payFlag || false}
           updateCloudCredential={
             controllerAPI.updateCloudCredential.bind(controllerAPI)}
+          updateAddress={
```

Comment:
Because there are so many methods I'm going to land in the camp of just passing `this.payment` through. It's nice to use the proptypes as a struct outlining which methods are necessary for the proper operation of the component though which is why we seem to always add the individual methods. This wouldn't be necessary if we could define structs for these instances :) TS maybe? :)